DOM Elements translation (#51)
reactjs/hu.reactjs.org: content/docs/reference-dom-elements.md

  - "tips/dangerously-set-inner-html.html"
---

> A React egy böngésző-független DOM rendszert implementál a teljesítmény és böngészők közti kompatibilitás érdekében. Éltünk a lehetőséggel és lekerekítettünk néhány érdes sarkot a DOM implementációban.

Reviewer: ```suggestion
A React egy böngészőfüggetlen DOM rendszert implementál a teljesítmény és böngészők közti kompatibilitás érdekében. Éltünk a lehetőséggel és lekerekítettünk néhány érdes sarkot a DOM implementációban.
```

> A Reactben minden DOM tulajdonság és attribútum (az eseménykezelőket beleértve) camelCase formában kell, hogy legyen írva. Például a `tabindex` HTML attribútum a `tabIndex`-szel egyenértékű Reactben. A kivétel az `aria-*` és a `data-*` attribútumok, amik kisbetűsek. Például az `aria-label`-t megarthatod `aria-label`-ként.

Reviewer: ```suggestion
A Reactben minden DOM tulajdonság és attribútum (az eseménykezelőket beleértve) camelCase formában kell, hogy legyen írva. Például a `tabindex` HTML attribútum a `tabIndex`-szel egyenértékű Reactben. A kivétel az `aria-*` és a `data-*` attribútumok, amik kisbetűsek. Például az `aria-label`-t megtarthatod `aria-label`-ként.
```

## Attribútumok közti különbségek {#differences-in-attributes}

Számos különbség van a React és a HTML attribútumainak működése között:

### checked {#checked}

A `checked` attribútum a `checkbox` vagy a `radio` típusú `<input>` komponens által támogatott. Használhatod arra, hogy azt jelöld egy komponens ki van-e választva, vagy sem. Ez hasznos kontrollált komponensek építése esetén. A `defaultChecked` kontrollálatlan megfelelője, ami a komponens létrejöttekor határozza meg, hogy az ki van-e választva, vagy sem.

### className {#classname}

Egy CSS osztály meghatározásához használd a `className` attribútumot. Ez minden általános DOM és SVG elemre vonatkozik, mint a `<div>`, `<a>` és a többi.

Ha a Reactet Web Komponensekkel használod (ami ritka), használd inkább a `class` attribútumot.

### dangerouslySetInnerHTML {#dangerouslysetinnerhtml}

> A `dangerouslySetInnerHTML` a React `innerHTML` helyettesítője a böngésző DOM-ban. Általában HTML-t így bevinni kockázatos, mert akarva-akaratlanul [cross-site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting) támadásnak teszed ki vele a felhasználóidat. Szóval annak ellenére, hogy bevihetsz HTML-t a Reacttel, ki kell írnod a `dangerouslySetInnerHTML` attribútumot, ami egy objektumot fogad egy `__htlm` tulajdonsággal, ami emlékeztessen arra, hogy ez veszélyes. Például:

Reviewer: ```suggestion
A `dangerouslySetInnerHTML` a React `innerHTML` helyettesítője a böngésző DOM-ban. Általában HTML-t így bevinni kockázatos, mert akarva-akaratlanul [cross-site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting) támadásnak teszed ki vele a felhasználóidat. Szóval annak ellenére, hogy bevihetsz HTML-t a Reacttel, ki kell írnod a `dangerouslySetInnerHTML` attribútumot, ami egy objektumot fogad egy `__html` tulajdonsággal, ami emlékeztessen arra, hogy ez veszélyes. Például:
```

### htmlFor {#htmlfor}

> Mivel a `for` egy JavaScriptnek fentartott szó, a React elemek helyette a `htmlFor`-t használják.

Reviewer: ```suggestion
Mivel a `for` egy JavaScriptnek fenntartott kulcsszó, a React elemek helyette a `htmlFor`-t használják.
```

### onChange {#onchange}

> Az `onChange` esemény úgy viselkedik, ahogy elvárnád: amikor egy űrlap mezője megváltozik, ez az esemény meg lesz hívva. Szándékosan nem használjuk a beépített böngésző-viselkedést, mivel az `onChange` egy helytelen elnevezés a viselkedésére, és a React erre az eseményre támaszkodik valósidejű felhasználói bevitelek kezelésére.

Reviewer: ```suggestion
Az `onChange` esemény úgy viselkedik, ahogy elvárnád: amikor egy űrlap mezője megváltozik, ez az esemény meg lesz hívva. Szándékosan nem használjuk a beépített böngészőviselkedést, mivel az `onChange` egy helytelen elnevezés a viselkedésére, és a React erre az eseményre támaszkodik valósidejű felhasználói bevitelek kezelésére.
```

### selected {#selected}

> A `selected` attribútumot az `<option>` komponens támogatja. Használhatod arra, hogy a komponens ki legyen-e választva, vagy sem. Ez hasznos amikor kontrollált komponenseket kell építeni.

Reviewer: ```suggestion
A `selected` attribútumot az `<option>` komponens támogatja. Használhatod arra, hogy a komponens ki legyen-e választva, vagy sem. Ez hasznos, amikor kontrollált komponenseket kell építeni.
```

};

function HelloWorldComponent() {
  return <div style={divStyle}>Helló világ!</div>;
}
```

Megjegyzendő hogy a stílusok nincsenek automatikusan prefixálva. Régebbi böngészők támogatásához szolgáltatnod kell a megfelelő tulajdonságokat:

```js
const divStyle = {
  WebkitTransition: 'all', // Figyeld meg a nagybetűs 'W'-t
  msTransition: 'all' // Az 'ms' az egyetlen vendor prefixum
};

function ComponentWithTransition() {
  return <div style={divStyle}>Ennek működnie kell böngészők között</div>;
}
```

A stíluskulcsok camelCase formátumban vannak, hogy konzisztens legyen az elérésük a DOM-on lévő csomópontokkal JS-ből. (pl. `node.style.backgroundImage`). A vendor prefixumok [kivétel az `ms`](https://www.andismith.com/blogs/2012/02/modernizr-prefixed/) nagybetűvel kell, hogy kezdődjenek. A `WebkitTransition`-nak ezért van nagy "W"-je.

> A React automatikusa hozzáilleszti a "px"-et bizonyos numerikus sorközi stílus tulajdonság után. Ha más egységet használnál a "px" helyett, az értéket sztringként add meg a kívánt egységgel. Például:

Reviewer: ```suggestion
A React automatikusan hozzáilleszti a "px"-et bizonyos numerikus sorközi stílustulajdonságok után. Ha más egységet használnál a "px" helyett, az értéket sztringként add meg a kívánt egységgel. Például:
```

```js
// Az stílus eredménye: '10px'
<div style={{ height: 10 }}>
  Helló világ!
</div>

// A stílus eredménye: '10%'
<div style={{ height: '10%' }}>
  Helló világ!
</div>
```

> Habár nem minden stílus tulajdonság van pixel sztringgé átkonvertálva. Néhányan egység nélkül maradnak (pl. `zoom`, `order`, `flex`). A teljes egység nélküli tulajdonságok listáját [itt](https://github.com/facebook/react/blob/4131af3e4bf52f3a003537ec95a1655147c81270/src/renderers/dom/shared/CSSProperty.js#L15-L59) találod.

Reviewer: ```suggestion
Habár nem minden stílustulajdonság van pixel sztringgé átkonvertálva. Néhányan egység nélkül maradnak (pl. `zoom`, `order`, `flex`). Az egység nélküli tulajdonságok teljes listáját [itt](https://github.com/facebook/react/blob/4131af3e4bf52f3a003537ec95a1655147c81270/src/renderers/dom/shared/CSSProperty.js#L15-L59) találod.
```

Ha szerver oldali React renderelést használsz, általában van egy figyelmeztetés, amikor a szerver és a kliens tartalma eltér. Azonban ritka esetekben nagyon nehéz, vagy lehetetlen a tökéletes egyezést garantálni. Például a időbélyegek várhatóan eltérnek a szerveren és a kliensen.

Ha a `suppressHydrationWarning`-ot `true`-ra állítod, a React nem fog figyelmeztetni az attribútumok, vagy az elem tartalmának eltérése esetén. Ez csak egy szint mélységig működik, és egy menekülési útnak van szánva. Ne használd túl sokat. A hidrálásról többet olvashatsz a [`ReactDOM.hydrate()` dokumentációban](/docs/react-dom.html#hydrate).

### value {#value}

A `value` attribútum az `<input>` és `<textarea>` komponensek által támogatott. Használhatod a komponens értéknek a megadására. Hasznos kontrollált komponensek készítéséhez. A `defaultValue` a kontrollálatlan változata, amivel a komponens értékét akkor tudod állítani, amikor az először jön létre.

## Minden támogatott HTML attribútum {#all-supported-html-attributes}

A React 16-tól kezdve minden standard, [vagy egyedi](/blog/2017/09/08/dom-attributes-in-react-16.html) DOM attribútum teljesen támogatott.

> A React mindigis biztosított egy JavaScript centrikus API-t a DOM-hoz. Mivel a React komponensek gyakran használnak mind egyedi és DOM-mal kapcsolatos propokat, a React konvenvió alapján `camelCase` formátumot használ úgy, ahogyan a DOM API-k:

Reviewer: ```suggestion
A React mindig is biztosított egy JavaScript centrikus API-t a DOM-hoz. Mivel a React komponensek gyakran használnak mind egyedi és DOM-mal kapcsolatos propokat, a React konvenció alapján `camelCase` formátumot használ úgy, ahogyan a DOM API-k:
```

```js

> <div tabIndex="-1" />      // Pont mint a node.tabIndex DOM API

Reviewer: ```suggestion
<div tabIndex="-1" />      // Pont, mint a node.tabIndex DOM API
```

> <div className="Button" /> // Pont mint a node.className DOM API

Reviewer: ```suggestion
<div className="Button" /> // Pont, mint a node.className DOM API
```

> <input readOnly={true} />  // Pont mint a node.readOnly DOM API

Reviewer: ```suggestion
<input readOnly={true} />  // Pont, mint a node.readOnly DOM API
```

```

> Ezek a propok nagyon hasonlóan működnek a megegyező HTML attribútumokhoz, az alábbi speciális esetek kivételével ami alább van dokumentálva.

Reviewer: ```suggestion
Ezek a propok nagyon hasonlóan működnek a megegyező HTML attribútumokhoz, a fentebb dokumentált speciális esetek kivételével.
```

> Néhány React által támoatott DOM attribútum:

Reviewer: ```suggestion
Néhány React által támogatott DOM attribútum:
```
